marathon/src/test/scala/io/buoyant/marathon/v2/AppIdNamerTest.scala

```diff
@@ -49,6 +49,24 @@ class AppIdNamerTest extends FunSuite with Awaits {
     assert(state == Activity.Ok(NameTree.Neg))
   }
 
+  test("Namer handles looking up both appId and /appId") {
+    class TestApi() extends Api {
+      def getAppIds(): Future[Api.AppIds] = {
+        Future.value(Set[String]("/foo", "/servicename"))
+      }
+      def getAddrs(app: String): Future[Set[SocketAddress]] = Future.never
+    }
+
+    val namer = new AppIdNamer(new TestApi(), Path.Utf8("io.l5d.marathon"), 250.millis)
+    @volatile var state: Activity.State[NameTree[Name]] = Activity.Pending
+
+    namer.lookup(Path(Buf.Utf8("servicename"), Buf.Utf8("residual"))).states respond { state = _ }
+    assert(state == Activity.Ok(NameTree.Leaf(Path(Buf.Utf8("io.l5d.marathon"), Buf.Utf8("servicename")))))
+
+    namer.lookup(Path(Buf.Utf8("/servicename"), Buf.Utf8("residual"))).states respond { state = _ }
+    assert(state == Activity.Ok(NameTree.Leaf(Path(Buf.Utf8("io.l5d.marathon"), Buf.Utf8("/servicename")))))
+  }
```

Comment:
Minor, but:

Ehh, I think your paths are funky here -- shouldn't have a slash within a path element otherwise the whole thing will end up hex-encoded:
```scala
com.twitter.finagle.Path.Utf8("/servicename").show ==
  """/\x2f\x73\x65\x72\x76\x69\x63\x65\x6e\x61\x6d\x65"""
```

I'd probably do `Path.read("/servicename/residual")` or `Path.Utf8("servicename", "residual")`.  Also I'd probably pull the paths out into temp vals so the assertions are a little more readable.